x-pack/plugins/task_manager/server/task_pool.ts

```diff
@@ -85,6 +85,10 @@ export class TaskPool {
     // this should happen less often than the actual changes to the worker queue
     // so is lighter than emitting the load every time we add/remove a task from the queue
     this.load$.next(asTaskManagerStatEvent('load', asOk(this.workerLoad)));
+    // cancel expired task whenever a call is made to check for capacity
+    // this ensures that we don't end up with a queue of hung tasks causing both
+    // the poller and the pool from hanging due to lack of capacity
+    this.cancelExpiredTasks();
```

Comment:
Would it give a more accurate representation of load if we cancelled the tasks before emitting the `load` stat?

Reply:
This did occur to me but canceling is actually an async operation as we all the task's `cancel` function and we don't wait on it.
This means that even if we flipped these two lines the value emitted to `load` would still be the one _before_ that task is actually canceled. :)

Reply:
Gotcha. So the call to `availableWorkers()` might not return the correct number of available workers either until the next poll interval, but that is better than never cancelling the expired tasks. Makes sense